plugins/vsphere/pkg/command/source/source.go

```diff
@@ -0,0 +1,90 @@
+/*
+Copyright 2020 VMware, Inc.
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package source
+
+import (
+	"net/url"
+	"time"
+
+	"github.com/spf13/cobra"
+	"knative.dev/pkg/apis"
+	duckv1 "knative.dev/pkg/apis/duck/v1"
+
+	"github.com/vmware-tanzu/sources-for-knative/plugins/vsphere/pkg"
+	"github.com/vmware-tanzu/sources-for-knative/plugins/vsphere/pkg/command"
+)
+
+type Options struct {
+	command.GenericOptions
+
+	Address       string
+	SkipTLSVerify bool
+	SecretRef     string
+
+	SinkURI        string
+	SinkAPIVersion string
+	SinkKind       string
+	SinkName       string
+
+	CheckpointMaxAge time.Duration
+	CheckpointPeriod time.Duration
+
+	PayloadEncoding string
+}
+
+func (so *Options) AsSinkDestination(namespace string) (*duckv1.Destination, error) {
+	apiURL, err := so.sinkURL()
+	if err != nil {
+		return nil, err
+	}
+	return &duckv1.Destination{
+		Ref: so.sinkReference(namespace),
+		URI: apiURL,
+	}, nil
+}
+
+func (so *Options) sinkURL() (*apis.URL, error) {
+	if so.SinkURI == "" {
+		return nil, nil
+	}
+	address, err := url.Parse(so.SinkURI)
+	if err != nil {
+		return nil, err
+	}
+	result := apis.URL(*address)
+	return &result, nil
+}
```

Comment:
Why not just use [`apis.ParseURL` here from Knative](https://pkg.go.dev/knative.dev/pkg@v0.0.0-20211213170220-c3a4928da77b/apis#ParseURL)? I think it does _exactly_ what you want.